drivers/input/input_internal.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2022, tangchunhui@coros.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "input_internal.h"
+
+int input_internal_setup(struct input_dev *dev)
```

Comment:
From one input device stand-point: are you going to call this more than once at runtime?

If not, then all of this can be integrated into a macro with relevant static intialization for each objects (rb, mutex, sem...)